pkg/streamingpromql/limiting/memory_consumption_test.go

```diff
@@ -44,6 +44,15 @@ func TestMemoryConsumptionTracker_Unlimited(t *testing.T) {
 	require.Equal(t, uint64(131), tracker.PeakEstimatedMemoryConsumptionBytes)
 
 	assertRejectedQueriesCount(t, reg, 0)
+
+	// Test reducing memory consumption to a negative panics
+	defer func() {
+		if r := recover(); r == nil {
+			t.Errorf("We did not panic on negative memory consumption")
+		}
+	}()
+
+	tracker.DecreaseMemoryConsumption(150)
```

Comment:
This could be simplified a bit with `require.Panics()`. 

(same comment applies below)